File: programmers-coding-test/C++/Level1/햄버거 만들기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> ingredient) {
    int answer = 0;
    
    string seq = "";
    for (int i : ingredient)
        seq += (i + '0');

    int pos = 3;
    while ((pos = seq.find("1231", max(0, pos - 3))) != string::npos) {
        seq = seq.substr(0, pos) + seq.substr(pos + 4);
        answer++;
    }

    return answer;
}

File: programmers-coding-test/Java/Level1/햄버거 만들기.java
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;

        StringBuilder seq = new StringBuilder();
        for (int i : ingredient)
            seq.append(i + '0');

        int index = 3;
        while ((index = seq.indexOf("1231", index - 3)) != -1) {
            seq.delete(index, index + 4);
            answer++;
        }

        return answer;
    }
}